lte/gateway/python/magma/mobilityd/uplink_gw.py

```diff
@@ -81,11 +82,11 @@ def get_gw_mac(self, vlan_id: Optional[str] = "") -> Optional[str]:
         else:
             return None
 
-    def update_mac(self, ip: str, mac: str, vlan_id: Optional[str] = ""):
+    def update_mac(self, ip: str, mac: Optional[str], vlan_id: Optional[str] = ""):
```

Comment:
you should push this nil-check to the caller. check for None at the same time you check for the empty string at the callsite you added/modified in this diff

Reply:
I have a patch for mac address validation improvements. I will take care of this in later patch.